### ISSUE TITLE
Programmatically generate ssl certs

### ISSUE THREAD

**Issue by cpacia** [OpenBazaar/openbazaar-go#868]
Create a `gencerts` command (cmd/gencerts.go) that will automatically generate self signed SSL certificates that can be used by the OpenBazaar electron client running on a remote machine. This is to make it easier for users to set up an instance on a remote machine and avoid manually typing openssl commands.

I gave it a go [here](https://github.com/OpenBazaar/openbazaar-go/pull/708) but could not get the client to accept the generated cert. 

@chaintip $50

**Comment by chaintip** [comment]
`0.03317627 BCH`| `~  51.21 USD` has [been tipped](https://explorer.bitcoin.com/bch/address/bitcoincash:qqh0ld4g6z2gkgefn9ls5sk35m0rdkudm5yv6wa52u) to this issue by @cpacia.
To claim [this bounty](http://github.chaintip.org/chaintip/bounties/issues/15), get a pull request merged with ```@chaintip fixes #868``` in the creation comment.

**Comment by tea-spoon** [comment]
Does someone have to build this into the existing codebase? Are you able to provide a small test folder with the function stubs that contain comments with the input/output you expect? This could then be moved into a functioning location within the repository later.

**Comment by cpacia** [comment]
Well the goal is to build it into the codebase but if you can make the functions work it shouldn't really be that hard to do it.

**Comment by ab10460ef3** [comment]
@cpacia, I refactored your code to fit with the changes in cmds.

https://github.com/ab10460ef3/openbazaar-go/tree/gencertsRSA

I had the same issue, client rejects self signed cert.  I think generating a CA and signing request might be necessary, or at least better received by clients. 

The only reason I switched to RSA is to make it as similar as possible to the current recommended commands.

If anyone wants to pickup the torch, everything that can be done with certificates is done here: 

https://github.com/cloudflare/cfssl
